erpnext/accounts/doctype/payment_entry/payment_entry.js

```diff
@@ -871,16 +871,25 @@ frappe.ui.form.on('Payment Entry', {
 			function(d) { return flt(d.amount) }));
 
 		if(frm.doc.party) {
-			if(frm.doc.payment_type == "Receive"
+			if(frm.doc.payment_type === "Receive"
 				&& frm.doc.base_total_allocated_amount < frm.doc.base_received_amount + total_deductions
 				&& frm.doc.total_allocated_amount < frm.doc.paid_amount + (total_deductions / frm.doc.source_exchange_rate)) {
-					unallocated_amount = (frm.doc.base_received_amount + total_deductions + frm.doc.base_total_taxes_and_charges
-						- frm.doc.base_total_allocated_amount) / frm.doc.source_exchange_rate;
-			} else if (frm.doc.payment_type == "Pay"
+					unallocated_amount = (
+						((frm.doc.base_received_amount + frm.doc.base_total_taxes_and_charges
+							 + (total_deductions > 0 ? total_deductions : 0))
+						/ frm.doc.source_exchange_rate)
+						- frm.doc.total_allocated_amount
+					);
+
+			} else if (frm.doc.payment_type === "Pay"
 				&& frm.doc.base_total_allocated_amount < frm.doc.base_paid_amount - total_deductions
```

Comment:
This is still some instance where this condition results in unallocated_amount = 0 even though there is still a balance remaining to be allocated.